test/LifToken.js

```diff
@@ -141,26 +146,43 @@ contract('LifToken', function(accounts) {
   it("should return correct balances after transferDataFrom and show the event on receiver contract", async function() {
     await help.simulateCrowdsale(token, 10000000, web3.toWei(0.1, 'ether'), [4000000,3000000,2000000,1000000,0], accounts);
     let message = await Message.new();
-    let messageReceiver = await Message.new();
-
-    let data = messageReceiver.contract.showMessage.getData(web3.toHex(123456), 666, 'Transfer Done');
+    help.abiDecoder.addABI(Message._json.abi);
 
-    await token.approve(message.contract.address, help.formatBalance(1000), {from: accounts[1]});
+    let data = message.contract.showMessage.getData(web3.toHex(123456), 666, 'Transfer Done');
 
-    let dataTransfer = token.contract.transferDataFrom.getData(accounts[1], messageReceiver.contract.address, help.formatBalance(1000), data, true);
+    await token.approve(accounts[2], help.formatBalance(1000), {from: accounts[1]});
 
-    let transaction = await message.call(token.contract.address, dataTransfer);
+    let transaction = await token.transferDataFrom(accounts[1], message.contract.address, help.formatBalance(1000), data, true, {from: accounts[2]});
+    let decodedEvents = help.abiDecoder.decodeLogs(transaction.receipt.logs);
 
-    assert.equal(2, transaction.receipt.logs.length);
-    assert.equal(data, '0x'+transaction.receipt.logs[1].data.substring(194, data.length+192));
-    assert.equal('0x1e24000000000000000000000000000000000000000000000000000000000000', transaction.logs[0].args.b32);
-    assert.equal(666, transaction.logs[0].args.number);
-    assert.equal('Transfer Done', transaction.logs[0].args.text);
-    assert.equal(help.formatBalance(1000), await token.balanceOf(messageReceiver.contract.address));
+    assert.equal(2, decodedEvents.length);
+    assert.equal(data, decodedEvents[1].events[3].value);
+    assert.equal('0x1e24000000000000000000000000000000000000000000000000000000000000', decodedEvents[0].events[0].value);
+    assert.equal(666, decodedEvents[0].events[1].value);
+    assert.equal('Transfer Done', decodedEvents[0].events[2].value);
+    assert.equal(help.formatBalance(1000), await token.balanceOf(message.contract.address));
 
     await help.checkValues(token, accounts,1000000, 10000000, 0, [3999000,3000000,2000000,1000000,0]);
   });
 
+  it("should return correct balances after approve and show the event on receiver contract", async function() {
```

Comment:
"correct balances" doesn't help me understand what's being asserted in the test. Maybe something like "unmodified balances after approve"? To make it more clear that they are not modified, should we have the same `checkValues` assertion before and after the call to approve?